core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -378,7 +289,8 @@ public List<TableIdentifier> listTables(SessionContext context, Namespace ns) {
               paths.tables(ns),
               queryParams,
               ListTablesResponse.class,
-              headers(context),
+              ImmutableMap.of(),
```

Comment:
It looks like we're not applying configured headers anymore? I don't think this was purely scoped to Auth headers.  Unless I'm missing it, we're not extracting environmental or property based headers.

Reply:
You've got a point here. I moved the config headers to OAuthManager but indeed, nothing implies that headers passed through configuration are specific to OAuth. I'm going to put them back 👍

Reply:
FYI I modified the tests today to include a custom header in every request. This was done in 
a0d97dc.